Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +22,70 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return ampHours_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if(currentNew_ < 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+int BatteryStateOfChargeService::getRemainingHours() const
+{
+    return remainingHours_;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return timeTillChargeOrDepletion_;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // This is where you can update your variables
-    // Hint: There are many different ways that the totalAmpHoursUsed can be updated
-    // i.e: Taking a running average of your data values, using most recent data points, etc.
+    QTime timeOld = timeNew_;
+    double currentOld = currentNew_;
+
+    currentNew_ = batteryData.current;
+    timeNew_ = batteryData.time;
+
+    if(timeOld.isNull()) // first itteration
+    {
+        ampHours_ = BATTERY_AMP_HOUR_CAPACITY * initialStateOfChargePercent_ / PERCENT_TO_DECIMAL;
```

Comment:
This looks like this should be updated in the initializer list, since it's starting value of `ampHours_`